schema/deploy/trigger_functions/create_or_refresh_review_step.sql

```diff
@@ -0,0 +1,47 @@
+-- Deploy ggircs-portal:trigger_functions/create_or_refresh_review_step to pg
+-- requires: tables/application_review_step
+
+begin;
+
+create or replace function ggircs_portal_private.create_or_refresh_review_step()
+  returns trigger as $$
+
+declare
+  temp_row record;
+
+begin
+
+  for temp_row in
+    select id, step_name from ggircs_portal.review_step where is_active = true
+  loop
+    if exists(select * from ggircs_portal.application_review_step where application_id = new.application_id and review_step_id = temp_row.id) then
+      update ggircs_portal.application_review_step set is_complete = false where application_id = new.application_id and review_step_id = temp_row.id;
+    else
+      insert into ggircs_portal.application_review_step(application_id, review_step_id)
+        values (new.application_id, temp_row.id);
+    end if;
```

Comment:
```suggestion
    insert into ggircs_portal.review_step(application_id, review_step_name_id) 
        values (new.application_id, temp_row.id) 
        on conflict (application_id, review_step_name_id) 
        do update set is_complete=false;
```
Would this work, and remove the need for the `if` statement?

Reply:
definitely, I've updated it